### PR TITLE
atlasaction: gitlab ci

### PR DESCRIPTION
**Pull request by noamcattan**

### CODE REVIEW

File: atlasaction/action.go [hunk 1]
```diff
@@ -48,6 +47,17 @@ type Action interface {
 	GetTriggerContext() (*TriggerContext, error)
 	// AddStepSummary adds a summary to the action step.
 	AddStepSummary(string)
+	// API returns as APIClient.
+	API() (APIClient, error)
+}
+
+type APIClient interface {
```
Comment (giautm) [inline atlasaction/action.go:54]: SCMClient or sth similar

Comment (a8m) [reply atlasaction/action.go:54]: Or just `SCM`.

File: atlasaction/gitlab_action.go [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+	}
+	defer f.Close()
+	f.WriteString(fmt.Sprintf("%s=%s\n", name, value))
+}
+
+// GetTriggerContext implements the Action interface.
+func (g *gitlabCI) GetTriggerContext() (*TriggerContext, error) {
+	ctx := &TriggerContext{
+		SCM: SCM{
+			Type:   atlasexec.SCMTypeGithub, // TODO: Change to Gitlab.
```
Comment (giautm) [inline atlasaction/gitlab_action.go:63]: Remind

File: atlasaction/gitlab_action.go [hunk 3]
```diff
@@ -0,0 +1,275 @@
...
+}
+
+// SetOutput implements the Action interface.
+func (g *gitlabCI) SetOutput(name, value string) {
+	f, err := os.OpenFile(".env", os.O_APPEND|os.O_WRONLY|os.O_CREATE, 0600)
+	if err != nil {
+		return
+	}
+	defer f.Close()
+	f.WriteString(fmt.Sprintf("%s=%s\n", name, value))
```
Comment (giautm) [inline atlasaction/gitlab_action.go:56]:
```suggestion
	fmt.Fprintf(f, "%s=%s\n", name, value)
```

**Review by giautm — COMMENTED** [review]
Look goods, please add testscript to ensure input/output are correct.

File: atlasaction/gitlab_ci_test.go [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+
+func TestGitlabCI(t *testing.T) {
+	testscript.Run(t, testscript.Params{
+		Dir: "testdata/gitlab",
+		Setup: func(env *testscript.Env) error {
+			commentsDir := path.Join(env.WorkDir, "comments")
+			if err := os.Mkdir(commentsDir, os.ModePerm); err != nil {
+				return err
+			}
+			srv := httptest.NewServer(newMockHandler(commentsDir))
```
Comment (giautm) [inline atlasaction/gitlab_ci_test.go:98]: Ensure cleanup the test server.

**Review by giautm — APPROVED** [review]
All goods, one mirror comment

**Pull request merged**